test/spmd/tensor/parallel/test_tp_style.py

```diff
@@ -170,6 +157,34 @@ def test_prepare_output_error(self):
         self._test_prepare_output_error(make_output_replicate_1d)
         self._test_prepare_output_error(make_output_tensor)
 
+    @with_comms
+    def test_rowwise_parallel_style(self):
+        tensor = torch.rand(8, 16, device=self.device_type)
+        rs = RowwiseParallel()
```

Comment:
Is this test passing?

Reply:
I need to change the baseclass init func. Not sure how it works on your end.

https://github.com/pytorch/tau/pull/623

Reply:
OK, I see the issue now. 

So if you do something like:

from spmd.tensor.parallel.style import RowwiseParallel. 

RowwiseParallel() works. 

But if we do from spmd.tensor.parallel import PairwiseParallel, you will get errors:
```
==================================== ERRORS ====================================
________ ERROR collecting test/spmd/tensor/parallel/test_tp_examples.py ________
test/spmd/tensor/parallel/test_tp_examples.py:18: in <module>
    from spmd.tensor.parallel import (
spmd/tensor/parallel/__init__.py:6: in <module>
    from spmd.tensor.parallel.api import (
spmd/tensor/parallel/api.py:151: in <module>
    parallel_style: ParallelStyle = PairwiseParallel(),
spmd/tensor/parallel/style.py:45: in __init__
    super().__init__(make_input_replicate_1d, make_output_tensor)
E   TypeError: object.__init__() takes exactly one argument (the instance to initialize)
=========================== short test summary info ============================
ERROR test/spmd/tensor/parallel/test_tp_examples.py - TypeError: object.__ini...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 2.80s ===============================
```

This is really hard to repro bug...

Reply:
Also you might want to rebase on top of https://github.com/pytorch/tau/pull/623. Thanks!